File: api/src/test/java/work/beltran/discogsbrowser/api/DiscogsServiceTest.java
package work.beltran.discogsbrowser.api;

import com.google.gson.Gson;

import org.junit.Before;

/**
 * Created by Miquel Beltran on 8/27/16
 * More on http://beltran.work
 */
public class DiscogsServiceTest {
    DiscogsService service;

    @Before
    public void setUp() throws Exception {
        String apiKey = System.getenv("DISCOGS_API_KEY");
        DiscogsServiceBuilderWithKey builder = new DiscogsServiceBuilderWithKey(apiKey, "Unit Test");
        Gson gson = GsonProvider.provideGson();
        service = builder.provideDiscogsService(gson);
    }
}

File: data-api/src/test/java/work/beltran/discogsbrowser/api/DiscogsServiceTest.kt
package work.beltran.discogsbrowser.api


import org.junit.Before

/**
 * Created by Miquel Beltran on 8/27/16
 * More on http://beltran.work
 */
abstract class DiscogsServiceTest {
    lateinit var service: DiscogsService

    @Before
    fun setUp() {
        val apiKey = System.getenv("DISCOGS_API_KEY")
        val builder = DiscogsServiceBuilderWithKey(apiKey, "Unit Test")
        service = builder.provideDiscogsService(MoshiProvider.provideMoshi())
    }

    companion object {
        internal const val USER = "mike513"
    }
}
